### ISSUE TITLE
[Transport client] TC returns "an exception occurred formatting a trace message"  when logging level is set to TRACE

### ISSUE THREAD

**Issue by narph** [elastic/elasticsearch#32742]
**Elasticsearch version** (`bin/elasticsearch --version`): 6.3.2

**Plugins installed**: [xpack]

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

When connecting to the cluster through the transport client we get the following in the elastic

```
2018-08-03 10:54:37.079 TRACE 11932 --- [ost-startStop-1] o.elasticsearch.indices.breaker.request  : [request] Adjusted breaker by [16440] bytes, now [16440]
2018-08-03 10:54:37.110 TRACE 11932 --- [ient_boss][T#1]] o.e.transport.netty4.ESLoggingHandler    : [id: 0x5b556509, L:/10.10.61.53:58833 - ...  ACTIVE
2018-08-03 10:54:37.110 TRACE 11932 --- [ient_boss][T#1]] o.e.transport.netty4.ESLoggingHandler    : an exception occurred formatting a trace message

java.io.EOFException: null
	at org.elasticsearch.common.io.stream.InputStreamStreamInput.readByte(InputStreamStreamInput.java:59) ~[elasticsearch-6.3.2.jar:6.3.2]
	at org.elasticsearch.common.io.stream.StreamInput.readString(StreamInput.java:378) ~[elasticsearch-6.3.2.jar:6.3.2]
	at org.elasticsearch.common.io.stream.StreamInput.readStringArray(StreamInput.java:465) ~[elasticsearch-6.3.2.jar:6.3.2]
	at org.elasticsearch.transport.netty4.ESLoggingHandler.format(ESLoggingHandler.java:109) ~[transport-netty4-client-6.3.2.jar:6.3.2]
	at org.elasticsearch.transport.netty4.ESLoggingHandler.format(ESLoggingHandler.java:50) ~[transport-netty4-client-6.3.2.jar:6.3.2]
	at io.netty.handler.logging.LoggingHandler.write(LoggingHandler.java:247) [netty-handler-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:816) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:723) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.ChannelDuplexHandler.write(ChannelDuplexHandler.java:106) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:738) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:730) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext.access$1900(AbstractChannelHandlerContext.java:38) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext$AbstractWriteTask.write(AbstractChannelHandlerContext.java:1081) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext$WriteAndFlushTask.write(AbstractChannelHandlerContext.java:1128) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.AbstractChannelHandlerContext$AbstractWriteTask.run(AbstractChannelHandlerContext.java:1070) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) [netty-common-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:404) [netty-common-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:464) [netty-transport-4.1.27.Final.jar:4.1.27.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884) [netty-common-4.1.27.Final.jar:4.1.27.Final]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
```

Logging set to:
```
logging.level.org.elasticsearch=TRACE
```
 

Dependency:
```
<dependency>
<groupId>org.apache.logging.log4j</groupId>
<artifactId>log4j-api</artifactId>
</dependency>
<dependency>
<groupId>org.apache.logging.log4j</groupId>
<artifactId>log4j-core</artifactId>
</dependency>
```

When changing the logging level, the exceptions disappear and the transport client is able to connect to the cluster.
cc @tlrx

**Comment by nikoncode** [comment]
Let me chance to look on that.

**Comment by tbrooks8** [comment]
I suspect this is related to this #27306.

**Comment by jasontedor** [comment]
Closed by #32725 and #33789
